src/logger.cpp

```diff
@@ -0,0 +1,38 @@
+#include "logger.h"
+
+QString currentDateTime() {
+    QDateTime date = QDateTime::currentDateTime();
+    QString dateTime = date.toString("dd.MM.yyyy hh:mm:ss");
+    return dateTime;
+}
+
+void Logger::log_error(QString msg) {
+    if(this->output_ == 0) {
+        qInfo() << currentDateTime() << this << ":" << msg; //how to print the class which is being logged
+                                                            //(i.e. the class emitting the signal/that is connected)?
+    } else {
+        QFile file(this->fileName_);
+        QTextStream stream(&file);
```

Comment:
In the event that the logger is logging to a file, we can probably save the stream itself as a member function and just keep writing to it that way